Remove PR comment restrictions from gh-safe.sh

yahgw-ai opened the pull request:
## Summary
- Removed the `is_pr_comment_fetch` function and associated checks from gh-safe.sh
- Cleaned up the script header to reflect the current functionality
- The gh-safe wrapper now only prevents merges and Claude/AI references

## Changes
- Deleted the `is_pr_comment_fetch` function that was checking for PR comment fetch commands
- Removed the conditional block that redirected PR comment fetching to collect-pr-comments.sh
- Removed PR comment content checking from the PR case statement
- Updated the script header comments to remove mention of PR comment redirection

no-issue: cleanup of internal tooling restrictions

yahgw-ai committed:
Remove PR comment restrictions from gh-safe.sh

Removed the is_pr_comment_fetch function and associated checks that were
redirecting PR comment fetching to the collect-pr-comments.sh script.
The gh-safe wrapper now only prevents merges and Claude/AI references.

yahgwai commented on scripts/gh-safe.sh:
```diff
@@ -54,56 +53,13 @@ is_merge_command() {
 }
 
-# Function to check if command is trying to fetch PR comments
-is_pr_comment_fetch() {
```
take a look at the setup.sh script, we have some regarding pr commetns to remove there too

yahgw-ai committed:
chore: remove PR comment reference from setup.sh

Removed the line mentioning "specific method for fetching github comments on PRs"
from the gh-safe wrapper description in the setup script.
